webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/ProjectExportService.scala

```diff
@@ -59,156 +56,102 @@ trait ProjectExportService {
  * @param graphIri the IRI of the named graph.
  * @param tempDir  the directory in which the file is to be saved.
  */
-private case class NamedGraphTrigFile(graphIri: IRI, tempDir: Path) {
+private case class NamedGraphTrigFile(graphIri: InternalIri, tempDir: Path) {
   lazy val dataFile: Path = {
-    val filename = graphIri.replaceAll("[.:/]", "_") + ".trig"
+    val filename = graphIri.value.replaceAll("[.:/]", "_") + ".trig"
     tempDir.resolve(filename)
   }
 }
 
-/**
- * A [[RdfStreamProcessor]] for combining several named graphs into one.
- *
- * @param formattingStreamProcessor a [[RdfStreamProcessor]] for writing the combined result.
- */
-private class CombiningRdfProcessor(formattingStreamProcessor: RdfStreamProcessor) extends RdfStreamProcessor {
-  private var startedStatements = false
-
-  // Ignore this, since it will be done before the first file is written.
-  override def start(): Unit = {}
+private object TriGCombiner {
 
-  // Ignore this, since it will be done after the last file is written.
-  override def finish(): Unit = {}
+  def combineTriGFiles(inputFiles: Iterable[Path], outputFile: Path): Task[Path] = ZIO.scoped {
+    for {
+      outFile   <- ZScopedJavaIoStreams.fileBufferedOutputStream(outputFile)
+      outWriter <- createPrefixDedupStreamRDF(outFile).map { it => it.start(); it }
+      _ <- ZIO.foreachDiscard(inputFiles)(file =>
+             // Combine the files and write the output to the given OutputStream
+             for {
+               is <- ZScopedJavaIoStreams.fileInputStream(file)
+               _  <- ZIO.attemptBlocking(RDFParser.source(is).lang(Lang.TRIG).parse(outWriter))
+             } yield ()
+           )
+    } yield outputFile
+  }
 
-  override def processNamespace(prefix: IRI, namespace: IRI): Unit =
-    // Only accept namespaces from the first graph, to prevent conflicts.
-    if (!startedStatements) {
-      formattingStreamProcessor.processNamespace(prefix, namespace)
-    }
+  private def createPrefixDedupStreamRDF(os: OutputStream): ZIO[Scope, Throwable, StreamRDF] = {
+    def acquire                    = ZIO.attempt(StreamRDFWriter.getWriterStream(os, Lang.TRIG))
+    def release(writer: StreamRDF) = ZIO.attempt(writer.finish()).logError.ignore
+    ZIO.acquireRelease(acquire)(release).map(dedupPrefixStream)
+  }
 
-  override def processStatement(statement: Statement): Unit = {
-    startedStatements = true
-    formattingStreamProcessor.processStatement(statement)
+  // Define a custom StreamRDF implementation to filter out duplicate @prefix directives
+  private def dedupPrefixStream(writer: StreamRDF): StreamRDF = new StreamRDFBase {
```

Comment:
The term "dedup" is short deduplication. Since this is a private method with a comment explaining it I was hoping that that would be clear enough.
https://en.wikipedia.org/w/index.php?title=Dedup&redirect=no